### PR TITLE
Add new CI jobs that test install/uninstall

### LINKED ISSUE #36: Add CI jobs to do Install -> Uninstall -> Install
Since the Github CI machines don't have Nix on them, we can just use them directly.

After the artifacts are produced, try to run `harmonic install $PLAN` and then do uninstall, then install again. This could be done on both x86 Linux and Darwin

### PR DESCRIPTION
**Pull request by Hoverbear**
Closes #36

### CODE REVIEW

File: src/planner/linux/multi.rs [hunk 1]
```diff
@@ -31,7 +31,7 @@ impl Planner for LinuxMulti {
                 Box::new(CreateDirectory::plan("/nix", None, None, 0o0755, true).await?),
                 Box::new(ProvisionNix::plan(self.settings.clone()).await?),
                 Box::new(ConfigureNix::plan(self.settings).await?),
-                Box::new(StartSystemdUnit::plan("nix-daemon.socket".into()).await?),
+                // Box::new(StartSystemdUnit::plan("nix-daemon.socket".into()).await?),
```
Comment (cole-h) [inline src/planner/linux/multi.rs:34]: Should this be deleted?

File: src/action/linux/start_systemd_unit.rs [hunk 2]
```diff
@@ -90,9 +90,13 @@ impl Action for StartSystemdUnit {
         tracing::debug!("Stopping systemd unit");
 
         // TODO(@Hoverbear): Handle proxy vars
-        execute_command(Command::new("systemctl").arg("stop").arg(format!("{unit}")))
-            .await
-            .map_err(|e| StartSystemdUnitError::Command(e).boxed())?;
+        execute_command(
+            Command::new("systemctl")
+                .arg("disable")
```
Comment (cole-h) [inline src/action/linux/start_systemd_unit.rs:95]: Is `disable` more forceful than `stop`? Or is this just so it doesn't get started again (by e.g. the socket)?

Comment (Hoverbear) [reply src/action/linux/start_systemd_unit.rs:95]: The `systemctl` docs state `disable` is the reverse: https://www.freedesktop.org/software/systemd/man/systemctl.html :woman_shrugging:

File: src/planner/linux/multi.rs (hunk 1 above)
Comment (Hoverbear) [reply src/planner/linux/multi.rs:34]: Yes!

**Review by cole-h — APPROVED** [review]
LGTM.

**Pull request merged**